assessment/report_generator.go

```diff
@@ -15,17 +15,147 @@
 package assessment
 
 import (
+	"encoding/csv"
 	"fmt"
+	"os"
 
 	"github.com/GoogleCloudPlatform/spanner-migration-tool/assessment/utils"
 	"github.com/GoogleCloudPlatform/spanner-migration-tool/logger"
 )
 
-func GenerateReport(assessmentOutput utils.AssessmentOutput) {
+type SchemaReportRow struct {
+	element          string
+	elementType      string // consider enum ?
+	sourceDefinition string
+	targetName       string
+	targetDefinition string
+	//DB
+	dbChangeType   string
+	dbChangeEffort string
+	dbImpact       string
+	//Code
+	codeChangeType    string // consider enum ?
```

Comment:
do we need a column for `Changes` as well?

Reply:
updated with changes